nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/operation/GetContainerRendition.java

```diff
@@ -0,0 +1,173 @@
+import org.nuxeo.ecm.automation.core.annotations.OperationMethod;
+import org.nuxeo.ecm.automation.core.annotations.Param;
+import org.nuxeo.ecm.automation.core.operations.blob.CreateZip;
+import org.nuxeo.ecm.automation.core.util.BlobList;
+import org.nuxeo.ecm.collections.api.CollectionManager;
+import org.nuxeo.ecm.collections.core.adapter.Collection;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelIterator;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.api.LifeCycleConstants;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.schema.FacetNames;
+import org.nuxeo.ecm.platform.rendition.Rendition;
+import org.nuxeo.ecm.platform.rendition.service.RenditionService;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Returns a Folderish Document or Collection default rendition.
+ *
+ * @since 9.3
+ */
+@Operation(id = GetContainerRendition.ID, category = Constants.CAT_BLOB, label = "Gets the folder's children or the collection's members default renditions", description = "Gets the list of blob of the folder's children or the collection's members default renditions. Returns a blob list file containing all the default rendition blobs.")
+public class GetContainerRendition {
+
+    private static final Log log = LogFactory.getLog(GetContainerRendition.class);
+
+    public static final String ID = "Document.GetContainerRendition";
+
+    @Context
+    protected RenditionService renditionService;
+
+    @Context
+    protected CollectionManager collectionManager;
+
+    @Param(name = "reason", required = false)
+    protected String reason;
+
+    @Param(name = "limit", description = "Limit of members to be returned. Default is no limit.", required = false)
+    protected int limit = -1;
+
+    @Param(name = "maxDepth", description = "Depth of the hierarchy to be explored. Default is 5.", required = false)
+    protected int maxDepth = 5;
+
+    @Param(name = "_currentDepth", required = false)
+    protected int currentDepth = 0;
+
+    @Context
+    protected CoreSession session;
+
+    protected int added = 0;
+
+    protected BlobList getCollectionBlobs(Collection collection) {
+        BlobList blobs = new BlobList();
+        for (String memberId : collection.getCollectedDocumentIds()) {
+            DocumentRef memberRef = new IdRef(memberId);
+            if (session.exists(memberRef)
+                    && !LifeCycleConstants.DELETED_STATE.equals(session.getCurrentLifeCycleState(memberRef))) {
+                DocumentModel member = session.getDocument(memberRef);
+                Blob blob = getDefaultRendition(member);
+                if (blob != null) {
+                    blobs.add(blob);
+                    if (limit > -1 && ++added >= limit) {
+                        break;
```

Comment:
Maybe log when the limit is reached to let people know that they might want to increase it.